ocf_data_sampler/load/open_tensorstore_zarrs.py

```diff
@@ -0,0 +1,92 @@
+"""Open multiple zarrs with TensorStore.
+
+This extendds the functionality of xarray_tensorstore to open multiple zarr stores
+"""
+
+import os
+
+import tensorstore as ts
+import xarray as xr
+from xarray_tensorstore import (
+    _raise_if_mask_and_scale_used_for_data_vars,
+    _TensorStoreAdapter,
+    _zarr_spec_from_path,
+)
+
+
+def tensorstore_open_multi_zarrs(
+    paths: list[str],
+    data_vars: list[str],
+    concat_axes: list[int],
+    context: ts.Context,
+    write: bool,
+) -> dict[str, ts.TensorStore]:
+    """Open multiple zarrs with TensorStore.
+
+    Args:
+        paths: List of paths to zarr stores.
+        data_vars: List of data variable names to open.
+        concat_axes: List of axes along which to concatenate the data variables.
+        context: TensorStore context.
+        write: Whether to open the stores for writing.
+    """
+    arrays_list = []
+    for path in paths:
+        specs = {k: _zarr_spec_from_path(os.path.join(path, k)) for k in data_vars}
+        array_futures = {
+          k: ts.open(spec, read=True, write=write, context=context)
+          for k, spec in specs.items()
+        }
+        arrays_list.append({k: v.result() for k, v in array_futures.items()})
```

Comment:
Ideally you might call `result()` in a separate loop, so all the arrays can be opened simultaneously.